src/qz/printer/action/PrintRaw.java

```diff
@@ -290,25 +288,19 @@ public void printJobRequiresAttention(PrintJobEvent printJobEvent) {
      */
     public void printToAlternate(PrintService service, byte[] cmds) throws IOException, PrintException {
         File tmp = File.createTempFile("qz_raw_", null);
-        printToFile(tmp, cmds);
-
-        String shellCmd = String.format("/usr/bin/lp -d \"%s\" -o raw \"%s\";", service.getName(), tmp.getAbsolutePath());
-        log.trace("Runtime Exec running: {}", shellCmd);
-
-        Process process = Runtime.getRuntime().exec(new String[] {"bash", "-c", shellCmd});
-        try { process.waitFor(); }catch(InterruptedException ignore) {}
-
-        BufferedReader buf = new BufferedReader(new InputStreamReader(process.getInputStream(), Charsets.UTF_8));
-        String output = IOUtils.toString(buf);
-        buf.close();
-
-        log.info("Runtime Exec returned: {}", output);
-        if (process.exitValue() != 0) {
-            throw new PrintException(String.format("Alternate printing returned a non-zero value (%d).%n%s", process.exitValue(), output));
-        }
-
-        if (!tmp.delete()) {
-            tmp.deleteOnExit();
+        try {
+            printToFile(tmp, cmds);
+            boolean success = ShellUtilities.execute(new String[] {
+                    "lp", "-d", PrintingUtilities.getPrinterId(service.getName()), "-o", "raw", tmp.getAbsolutePath()
+            });
+
+            if (!success) {
+                throw new PrintException("Alternate printing failed.  Please troubleshoot the above commands manually.");
```

Comment:
Isn't this statement going to look out of context when a stacktrace is actually thrown?